unit_tests/testStrings/testStrings.f90

```diff
@@ -589,7 +589,75 @@ SUBROUTINE testStrFunct()
+    str_list = str%rPartition('')
+    ASSERT_EQ(SIZE(str_list),3,'non-null string / null delimiter')
+    ASSERT_EQ(CHAR(str_list(1)),'','non-null string / null delimiter')
+    ASSERT_EQ(CHAR(str_list(2)),'','non-null string / null delimiter')
+    ASSERT_EQ(CHAR(str_list(3)),'abc','non-null string / null delimiter')
+
+    str = ''
+    str_list = str%partition('---')
+    ASSERT_EQ(SIZE(str_list),3,'null string / non-null delimiter')
+    ASSERT_EQ(CHAR(str_list(1)),'','null string / non-null delimiter')
+    ASSERT_EQ(CHAR(str_list(2)),'---','null string / non-null delimiter')
+    ASSERT_EQ(CHAR(str_list(3)),'','null string / non-null delimiter')
+    str_list = str%rPartition('---')
+    ASSERT_EQ(SIZE(str_list),3,'null string / non-null delimiter')
+    ASSERT_EQ(CHAR(str_list(1)),'','null string / non-null delimiter')
+    ASSERT_EQ(CHAR(str_list(2)),'---','null string / non-null delimiter')
+    ASSERT_EQ(CHAR(str_list(3)),'','null string / non-null delimiter')
+
+    str = 'abc'
+    str_list = str%partition('---')
+    ASSERT_EQ(SIZE(str_list),3,'string, delimiter not found')
+    ASSERT_EQ(CHAR(str_list(1)),'abc','string, delimiter not found')
+    ASSERT_EQ(CHAR(str_list(2)),'---','string, delimiter not found')
+    ASSERT_EQ(CHAR(str_list(3)),'','string, delimiter not found')
+    str_list = str%rPartition('---')
+    ASSERT_EQ(CHAR(str_list(1)),'','string, delimiter not found')
+    ASSERT_EQ(CHAR(str_list(2)),'---','string, delimiter not found')
+    ASSERT_EQ(CHAR(str_list(3)),'abc','string, delimiter not found')
+
+    str = 'abc---def'
+    str_list = str%partition('---')
+    ASSERT_EQ(SIZE(str_list),3,'string with single delimter found')
+    ASSERT_EQ(CHAR(str_list(1)),'abc','string with single delimter found')
+    ASSERT_EQ(CHAR(str_list(2)),'---','string with single delimter found')
+    ASSERT_EQ(CHAR(str_list(3)),'def','string with single delimter found')
+    str_list = str%rPartition('---')
+    ASSERT_EQ(SIZE(str_list),3,'string with single delimter found')
+    ASSERT_EQ(CHAR(str_list(1)),'abc','string with single delimter found')
+    ASSERT_EQ(CHAR(str_list(2)),'---','string with single delimter found')
+    ASSERT_EQ(CHAR(str_list(3)),'def','string with single delimter found')
+
+    str = 'abc---def---ghi'
+    str_list = str%partition('---')
+    ASSERT_EQ(SIZE(str_list),3,'string with multiple delimter found')
+    ASSERT_EQ(CHAR(str_list(1)),'abc','string with multiple delimter found')
+    ASSERT_EQ(CHAR(str_list(2)),'---','string with multiple delimter found')
+    ASSERT_EQ(CHAR(str_list(3)),'def---ghi','string with multiple delimter found')
+    str_list = str%rPartition('---')
+    ASSERT_EQ(SIZE(str_list),3,'string with multiple delimter found')
+    ASSERT_EQ(CHAR(str_list(1)),'abc---def','string with multiple delimter found')
+    ASSERT_EQ(CHAR(str_list(2)),'---','string with multiple delimter found')
+    ASSERT_EQ(CHAR(str_list(3)),'ghi','string with multiple delimter found')
+
```

Comment:
Could you also add tests for 2 more cases: when the delimiter starts the string, and when the delimiter ends the string?  I think it should be fine, but just want to make sure there are no indexing gotchas